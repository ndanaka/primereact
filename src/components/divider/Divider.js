import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/ClassNames';

export class Divider extends Component {

    static defaultProps = {
        align: null,
        layout: 'horizontal',
        type: 'solid',
        style: null,
        className: null
    }

    static propTypes = {
        align: PropTypes.string,
        layout: PropTypes.string,
        type: PropTypes.string,
        style: PropTypes.object,
        className: PropTypes.string
    };

    render() {
        const dividerClassName = classNames('p-divider p-component', 'p-divider-' + this.props.layout, 'p-divider-' + this.props.type,
            { 'p-divider-left': this.props.layout === 'horizontal' && (!this.props.align || this.props.align === 'left') },
            { 'p-divider-center': this.props.layout === 'horizontal' && this.props.align === 'center' },
            { 'p-divider-right': this.props.layout === 'horizontal' && this.props.align === 'right' },
            { 'p-divider-top': this.props.layout === 'vertical' && (this.props.align === 'top') },
            { 'p-divider-center': this.props.layout === 'vertical' && (!this.props.align || this.props.align === 'center') },
            { 'p-divider-bottom': this.props.layout === 'vertical' && this.props.align === 'bottom' },
            this.props.className);

        return (
            <div className={dividerClassName} style={this.props.style} role="separator">
                <div className="p-divider-content">
                    {this.props.children}
                </div>
            </div>

        );
    }
}
