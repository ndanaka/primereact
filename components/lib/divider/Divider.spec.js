import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Divider } from './Divider';

const snapshot = (props, name) => expect(render(<Divider {...props} />).container).toMatchSnapshot(name);

describe('Divider Snapshot', () => {
    test('check snapshots', () => {
        snapshot({}, 'default');
        snapshot({ layout: 'horizontal', align: 'left' }, 'horizontal left');
        snapshot({ layout: 'horizontal', align: 'right' }, 'horizontal right');
        snapshot({ layout: 'horizontal', align: 'center' }, 'horizontal center');
        snapshot({ layout: 'vertical', align: 'center' }, 'vertical center');
        snapshot({ layout: 'vertical', align: 'top' }, 'vertical top');
        snapshot({ layout: 'vertical', align: 'bottom' }, 'vertical bottom');
        snapshot({ type: 'dashed' }, 'dashed');
        snapshot({ type: 'dotted' }, 'dotted');
    });
});
