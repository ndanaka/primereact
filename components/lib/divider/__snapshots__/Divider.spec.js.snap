// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Snapshot check snapshots: dashed 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-dashed p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: default 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: dotted 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-dotted p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: horizontal center 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-center"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: horizontal left 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: horizontal right 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-right"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: vertical bottom 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-bottom"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: vertical center 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-center"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider Snapshot check snapshots: vertical top 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-top"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;
